packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -530,27 +529,24 @@ abstract class FlameChartState<T extends FlameChart,
     @required int startMicros,
     @required int durationMicros,
     @required V data,
-    bool scrollHorizontally = true,
     bool scrollVertically = true,
     bool jumpZoom = false,
   }) async {
-    await Future.wait([
-      zoomToTimeRange(
-        startMicros: startMicros,
-        durationMicros: durationMicros,
-        jump: jumpZoom,
-      ),
-      if (scrollVertically) scrollVerticallyToData(data),
-    ]);
-    // Call this in a post frame callback so that the horizontal scroll
-    // controllers have had time to update their scroll extents. Otherwise, we
-    // can hit a race where are trying to scroll to an offset that is beyond
-    // what the scroll controller thinks its max scroll extent is.
-    if (scrollHorizontally) {
-      WidgetsBinding.instance.addPostFrameCallback((_) async {
-        await scrollHorizontallyToData(data);
-      });
-    }
+    await zoomToTimeRange(
+      startMicros: startMicros,
+      durationMicros: durationMicros,
+      jump: jumpZoom,
+    );
+    // Call these in a post frame callback so that the scroll controllers have
+    // had time to update their scroll extents. Otherwise, we can hit a race
+    // where are trying to scroll to an offset that is beyond what the scroll
+    // controller thinks its max scroll extent is.
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      await Future.wait([
+        scrollHorizontallyToData(data),
+        if (scrollVertically) scrollVerticallyToData(data),
+      ]);
+    });
```

Comment:
why does zooToTimeRange impact the vertical scroll controller? I would think that would only impact horizontal.

Reply:
It doesn't impact the vertical scroll controller, but when we call scrollVerticallyToData(data), that triggers a rebuild that may occur before all the horizontal scroll controllers have updated.